text/3537-msrv-resolver.md

```diff
@@ -0,0 +1,879 @@
+
+Compared to the above workflow, this is likely targeted at just library and tool maintainers as other use cases don't have access to the latest version or they are needing the repo to be compatible with their MSRV.
+
+**Priority 4 because:**
+- The MSRV has various carve outs, providing an inconsistent experience compared to other packages using other workflows and affecting the quality of the ecosystem
+  - For workspaces with bins, `cargo install --locked` is expected to work with the MSRV but won't
+  - If they use new Cargo features, then `[patch]`ing in a git source for the dependency won't work
+  - For contributors, they must be on an unspecified Rust toolchain version
+- The caveats involved in this approach (see prior item) would lead to worse documentation which lowers the quality to users
+- This still leads to stagnation despite being able to use the latest dependencies as they are limited in what they can use from them and they can't use features from the latest Rust toolchain
+- These library and tool maintainers are absorbing costs from the less common use cases of their dependents
+  - They could shift these costs to those that need old versions by switching to the "Latest MSRV" workflow by allowing their users to backport fixes to prior MSRV releases
+
+**Pain points:**
+
+Like the prior workflow, when developing with the latest toolchain,
+feedback is delayed until CI or an embedded image build process which can be frustrating
+(using too-new dependencies, using too-new Cargo or Rust features, etc).
+
+When `Cargo.lock` is resolved for latest dependencies, independent of MSRV,
+verifying the MSRV becomes difficult as they  must either juggle two lockfiles, keeping them in sync, or use the unstable `-Zminimal-versions`.
+The two lockfile approach also has all of the problems shown earlier in writing the lockfile.
+
+When only keeping MSRV-incompatible `dev-dependencies`,
+one lockfile can be used but it can be difficult to edit the `Cargo.lock` to ensure you get new `dev-dependencies` without infecting other dependency types.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## The `rust-version` field
+
+*(update to [manifest documentation](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field))*
+
+The `rust-version` field is an optional key that tells cargo what version of the
+Rust language and compiler your package can be compiled with. If the currently
+selected version of the Rust compiler is older than the stated version, cargo
+will exit with an error, telling the user what version is required.
+To support this, Cargo will prefer dependencies that are compatible with your `rust-version`.
+
+The first version of Cargo that supports this field was released with Rust 1.56.0.
+In older releases, the field will be ignored, and Cargo will display a warning.
+
+```toml
+[package]
+# ...
+rust-version = "1.56"
+```
```

Comment:
Are we saying that `rust-version = "1.56.0"` behaves like `rust-version =  "=1.56.0"`, if we translate it to `cargo`'s semver modifiers?

This would be quite a surprise. Why shouldn't it be `^1.56.0`, as we currently do for package dependencies in `Cargo.toml`?